File: sem2/python/insertsort.py
#!/usr/bin/python3

def insertion_sort(a):
	for i in range(1,len(a)):
		key = a[i]
		j = i
		while (key < a[j-1] and j > 0):
			a[j] = a[j-1]
			j = j -1
		a[j] = key
	return a

a = list(map(int, input("Enter contents of array: ").split()))
print("Sorted array:", *insertion_sort(a))

File: sem2/cpp/insertsort.cpp
#include <bits/stdc++.h>

using namespace std;

void insertsort(int a[], int n) {
	for(int i=1; i<n; i++) {
		int k = i;
		while(a[k] < a[k-1]) {
			swap(a[k], a[k-1]);
			k--;
		}
	}
}

int main() {
	cout << "Enter the size of array:\n";
	int n;
	cin >> n;
	cout << "Enter contents of array:\n";
	int a[n];
	for(int i=0; i<n; i++)
		cin >> a[i];
	insertsort(a,n);
	cout << "Sorted array:\n";
	for(int i=0; i<n; i++) cout << a[i] << " ";
	cout << endl;
	return 0;
}
